Miscellaneous PEP8 and minor changes

brbsix opened the pull request:
I have a few minor changes in case you're interested.
- PEP8 stuff (indentation, whitespace, unused objects, reorder imports)
- Logging messages are now output as follows (to differentiate between the various types of message):

> ERROR: Unable to connect to 'www.speedtest.net'
> INFO: Took 15559 ms to upload 1255152 bytes
- The program exits with an exit status of 1 upon error (like any other *nix program). Note use of `sys.exit(1)` rather than `exit(1)`. See https://bugs.python.org/issue8220.
- Also I noticed a prior screw-up of mine. `'%d' % random.random()` is `'0'`, which is obviously not the desired result. So that's been fixed and replaced with the creation of a random 12-digit integer.
- Fix for #7. Test for expected ping output and ensure pipeline exits successfully.

brbsix committed:
Simplify --version option

brbsix committed:
Remove trailing whitespace

brbsix committed:
Add space

brbsix committed:
Fix import order (PEP8)

brbsix committed:
Exit program with an exit code of 1 if error is encountered

brbsix committed:
Remove unused __script__

brbsix committed:
Import print function

Shuts up the following pylint warning:
> Unnecessary parens after u'print' keyword (superfluous-parens)

brbsix committed:
Fix bad indentation

brbsix committed:
Show logging level name with messages

brbsix committed:
Adjust connect's error message (avoid superfluous verbiage)

brbsix committed:
Fix random digit integer strings for use in urls

brbsix committed:
Include expected ping output in test

brbsix committed:
Ensure all pipeline commands exit successfully

Fixes PR #7

brbsix committed:
Format exception messages with repr() instead of str() (shows quotes)

sirgrigio commented:
@fopina Any chance to see this one merged? :)

Also, @brbsix, I was considering some more little things: 
1. we are actually logging on root, we would need something like `log = logging.getLogget('pyspeedtest')` to differentiate the log output; 
2. inside the `chooserver()` method the log statement about ip and coordinates goes on three different lines, whereas it should probably be on a single line or splitted into three different log statements;
3. every log statement is on info level but I think it would be better to differentiate between info and debug.

I was going to make these little changes in another PR but then I realised that it is better if you include them in this one perhaps?!

brbsix committed:
Split up chooseserver() log message

brbsix committed:
Differentiate between info and debug-level log messages

brbsix committed:
Don't use the root logger (to avoid conflicts when used as a module)

The program's logger is declared in the global namespace as `LOG`.
`logging.basicConfig()` is only helpful on the root logger, so instead
perform the logger configuration in `init_logging()`.

brbsix commented:
@zmcode How does that look?

sirgrigio commented:
Great! I like it 👍

brbsix committed:
Use list comprehensions to create connections

brbsix committed:
Optimize generation of upload data

Profiling the code revealed significant time spent in `random.choice()`.
As such, an alternative was implemented. See the improvement:

>>> time with_random_choice()
CPU times: user 668 ms, sys: 4 ms, total: 672 ms
Wall time: 790 ms

>>> time without_random_choice()
CPU times: user 108 ms, sys: 0 ns, total: 108 ms
Wall time: 110 ms

sirgrigio commented:
@fopina sorry if I insist on it but is there a way we can see this one merged? Perhaps would you prefer it to be split into smaller and more homogeneous requests?

fopina commented:
Done! Thanks
